### DISCUSSION TITLE
Issue with getFillColor transition in layer

### DISCUSSION THREAD

**Discussion by j3lev** [visgl/deck.gl#5283] [Q&A] [answered]
Hey there, really appreciate all of the hard work that's been put into this library. I'm having an issue with getting transitions to work with a layer. Here is a snippet of my code:
```
setLayers([
          new H3HexagonLayer({
            updateTriggers: { getFillColor: props.forecastIndex },
            data: props.data?.hexGeo,
            stroked: true,
            extruded: false,
            getHexagon: d => d.hex,
            getFillColor: d => [...getColor(d.interaction * factor).rgb(), 150],
            transitions: {
              getFillColor: {
                type: 'interpolation',
                duration: 500
              }
            }
          })
        ])
```
The fill color updates without an issue but there is no transition. I've tried this with easing and enter props with no luck. Could someone please point me in the right direction? I'm using v8.3.11

**Comment by Pessimistress** [comment, marked as answer]
I think this is a bug in the H3HexagonLayer. It is not forwarding transition props to its sub layer.

**Reply by j3lev** [reply]
Ah interesting, is there something i can do explicitly to forward the props or should I just convert the data to GeoJSON and get rid of the H3 layer?

**Reply by j3lev** [reply]
Also should I submit a bug for this?

**Reply by Pessimistress** [reply]
Yes if you don't mind!

**Reply by Pessimistress** [reply]
> is there something i can do explicitly to forward the props

As a temporary work around, you can specify [_subLayerProps](https://deck.gl/docs/api-reference/core/composite-layer#_sublayerprops) of the H3HexagonLayer with your transitions:

```js
new H3HexagonLayer({
  _subLayerProps: {
    'hexagon-cell': {
      transitions: {
         getFillColor: 500
      }
    }
  }
})
```

Sub layer keys can be found [here](https://deck.gl/docs/api-reference/geo-layers/h3-hexagon-layer#sub-layers)
